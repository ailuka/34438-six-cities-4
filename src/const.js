export const EstateType = {
  APARTMENT: `apartment`,
  HOUSE: `house`,
  HOTEL: `hotel`,
  ROOM: `room`,
};

export const CityName = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const MapClass = {
  MAP_MAIN: `cities__map`,
  MAP_DETAILS: `property__map`,
};

export const OfferClassNamesForPageType = {
  main: {
    article: `cities__place-card`,
    image: `cities__image-wrapper`,
    list: `cities__places-list tabs__content`,
    info: ``,
  },
  details: {
    article: `near-places__card`,
    image: `near-places__image-wrapper`,
    list: `near-places__list`,
    info: ``,
  },
  favorites: {
    article: `favorites__card `,
    image: `favorites__image-wrapper`,
    list: ``,
    info: `favorites__card-info`,
  }
};

export const KeyCodes = {
  ENTER: 13,
  SPACEBAR: 32,
};

export const SortTypes = {
  POPULAR: 1,
  PRICE_LOW_TO_HIGH: 2,
  PRICE_HIGH_TO_LOW: 3,
  TOP_RATED_FIRST: 4,
};

export const AppRoute = {
  ROOM: `/offer/:id`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`,
  ROOT: `/`,
};
