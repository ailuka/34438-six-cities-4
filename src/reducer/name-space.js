const NameSpace = {
  ERROR: `ERROR`,
  FILTERS: `FILTERS`,
  OFFERS: `OFFERS`,
  USER: `USER`,
};

export {NameSpace};
