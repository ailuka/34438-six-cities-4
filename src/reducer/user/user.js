import {extend} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  email: null,
};

const ActionType = {
  CHANGE_AUTH_STATUS: `CHANGE_AUTH_STATUS`,
  SET_EMAIL: `SET_EMAIL`,
};

const ActionCreator = {
  changeAuthStatus: (status) => {
    return {
      type: ActionType.CHANGE_AUTH_STATUS,
      payload: status,
    };
  },
  setEmail: (email) => {
    return {
      type: ActionType.SET_EMAIL,
      payload: email,
    };
  },
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.changeAuthStatus(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setEmail(response.data.email));
      });
    // .catch((err) => {
    //   throw err;
    // });
  },

  logIn: (userCredentials) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: userCredentials.login,
      password: userCredentials.password,
    })
      .then((response) => {
        dispatch(ActionCreator.changeAuthStatus(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setEmail(response.data.email));
      });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_AUTH_STATUS:
      return extend(state, {
        authorizationStatus: action.payload,
      });
    case ActionType.SET_EMAIL:
      return extend(state, {
        email: action.payload,
      });
  }

  return state;
};

export {AuthorizationStatus, ActionType, ActionCreator, Operation, reducer};
