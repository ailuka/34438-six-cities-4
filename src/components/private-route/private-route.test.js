import renderer from "react-test-renderer";
import React from "react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {MemoryRouter as Router} from "react-router";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {AppRoute} from "../../const.js";
import {NameSpace} from "../../reducer/name-space.js";
import PrivateRoute from "./private-route.jsx";

const MockComponent = () => <div />;
const mockStore = configureStore([]);

describe(`PrivateRoute snapshot`, () => {
  it(`should not render MockComponent`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
        email: ``,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router initialEntries={[AppRoute.FAVORITE]}>
            <PrivateRoute
              exact
              path={AppRoute.FAVORITE}
              render={() => {
                return <MockComponent/>;
              }}
            />
          </Router>
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render MockComponent`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        email: `abc@gmail.com`,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router initialEntries={[AppRoute.FAVORITE]}>
            <PrivateRoute
              exact
              path={AppRoute.FAVORITE}
              render={() => {
                return <MockComponent/>;
              }}
            />
          </Router>
        </Provider>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
