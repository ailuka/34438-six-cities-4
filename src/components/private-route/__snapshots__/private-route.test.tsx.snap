// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute snapshot should not render MockComponent 1`] = `null`;

exports[`PrivateRoute snapshot should render MockComponent 1`] = `<div />`;
